activerecord/test/cases/transactions_test.rb

```diff
@@ -954,27 +954,25 @@ def test_no_automatic_savepoint_for_inner_transaction
   end
 end if Topic.connection.supports_savepoints?
 
-if current_adapter?(:PostgreSQLAdapter, :Mysql2Adapter)
+unless current_adapter?(:SQLite3Adapter)
```

Comment:
Prefer whitelist. Like 3efad391e6bcdab2b86aacc45b9a7ade509e6c4f, cb2934ed79286f46c1e5593dd4c34aebfb2bc4d6.

Reply:
Thanks for the review. Updated to use a white list.

Reply:
I think black list on this one is fine. The number of adapters than doesn't support is less than the number of adapters that supports. I'd say, maybe SQLite is the only adapter we will even skip.

Reply:
Appreciate to get two reviews, but unfortunately, they conflict. I have pushed another commit to use the blacklist, whichever decision made by committers I'll squash or revert.

Reply:
How about `if ActiveRecord::Base.connection.supports_transaction_isolation?`?

I also thought that most adapters would pass the tests, but basically I had used conservative whitelist unless actually confirmed that (of course 89603b41edbaaed5cab4964a30caa8bda9285879 too).
In this case, I re-thought (and realized), if any RDBMS supports transaction isolation, the default isolation level should be expected to be more than read commited.

Reply:
If possible checking for the feature we want I think we should

Reply:
> How about if ActiveRecord::Base.connection.supports_transaction_isolation??

I think it make sense. Especially `test_transaction_isolation__read_committed` is testing if the transaction supports read committed. Since MySQL with InnoDB storage engine the default isolation level is repeatable read(more strict than read commiteed). Oracle and PostgreSQL default one is read committed. Both of them support read committed, do not allow dirty read.